### PR TITLE
[FeatureHighlight] Add a voiceover dismissal affordance for the feature highlight.

### LINKED ISSUE #8961: Feature highlight doesn't provide dismissal affordance in voiceover unless the action is taken.
Currently with Feature highlight, we have a hint that is spoken on the title, to double tap to dismiss, but you cannot double tap to dismiss. The only way to dismiss is to Z escape or take the action forcefully.

Acceptance Criteria:
* Hint of double tap for dismissal should not be spoken on the title as no action to dismiss can be done on the title.
* An affordance is provided to dismiss the feature highlight once it is presented for voiceover users.

---
#### Internal data
- Associated internal bug: [b/144572543](http://b/144572543)

### PR DESCRIPTION
**Pull request by yarneo**
Currently the feature highlight didn't have a proper dismiss affordance when voiceover is on.

It will instead speak a hint of "double tap to dismiss" on the title but without allowing the user to dismiss even when double tapped.

This adds a dismiss affordance and passes the accessibility hint to dismiss to that view.

Frame of dismissal button in VO:
IMG_2140

Closes #8961 #8888 #8884 #8886 #8885

### CODE REVIEW

**Comment by jverkoey** [comment]
Please point to the GitHub issue URL in the PR description.

**Comment by yarneo** [comment]
It isn't populated yet.

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m [hunk 1]
```diff
@@ -108,6 +109,12 @@ - (instancetype)initWithFrame:(CGRect)frame {
     _displayMaskLayer = [[MDCFeatureHighlightLayer alloc] init];
     _displayMaskLayer.fillColor = [UIColor whiteColor].CGColor;
 
+    _accessibilityView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 1, 1)];
```
Comment (jverkoey) [inline components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m]: Should this frame be the entire feature highlight in some manner?

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m [hunk 2]
```diff
@@ -108,6 +109,12 @@ - (instancetype)initWithFrame:(CGRect)frame {
...
+    _accessibilityView.isAccessibilityElement = YES;
+    _accessibilityView.accessibilityTraits = UIAccessibilityTraitButton;
+    _accessibilityView.accessibilityLabel = @"Dismiss highlight button";
+    [self addSubview:_accessibilityView];
```
Comment (jverkoey) [inline components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m:118]: Did you explore making `self` be the a11y element?

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m (hunk 1 above)
Comment (yarneo) [reply components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m]: it is with its accessibility frame. I didn't want it to cause the entire tappable area for non voiceover users to become dismissable and thus affect them

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m (hunk 2 above)
Comment (yarneo) [reply components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m:118]: I did, but with that it caused more complications as then its subviews won't be accessible, and then I would have to essentially change the hierarchy and add a parentView on top of self causing other complications with the controller searching for a MDCFeatureHighlightView as its base view for animation and such

**Comment by jverkoey** [comment]
Please link to the GitHub issue URL in the PR description.

**Comment by yarneo** [comment]
Linked

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m (hunk 2 above)
Comment (jverkoey) [reply components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m:118]: Gotcha; thank you for the context.

**Review by jverkoey — COMMENTED** [review]
I can't say with confidence that this approach is an improvement overall and I'm not certain what an improvement might be here without digging in, so I'll resign from reviewers on this for now.

**Review by romoore — APPROVED** [review]
LGTM. Please add a snapshot with VoiceOver showing the frame of the newly-added view.

**Comment by jverkoey** [comment]
Note this PR caused a test breakage and is being rolled back in: https://github.com/material-components/material-components-ios/pull/9015

**Pull request merged**